### PR TITLE
feat: Animate Page Transitions and Metadata display pages

### PR DESCRIPTION
**Pull request by joannalikespi**
Full implementation of animated page transitions and metadata pages.
Includes:
-Animated transitions between the configure recording set and the preview metadata pages
-Implementation of the dvcfg, oto.ini, and reclist.txt pages

### CODE REVIEW

File: src/renderer/components/configure-recording-set-page/index.scss [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+.slide-right-enter {
+  opacity: 0;
+  transform: translateX(-100%);
+}
+.slide-right-enter-active {
+  opacity: 1;
+  transform: translateX(0%);
+  transition: opacity 300ms, transform 300ms;
+}
+.slide-right-appear {
```
Comment (iluminar-yi) [inline src/renderer/components/configure-recording-set-page/index.scss:41]: New line above?

File: src/renderer/components/configure-recording-set-page/configure-recording-set.tsx [hunk 2]
```diff
@@ -0,0 +1,306 @@
+import log from 'electron-log';
```
Comment (iluminar-yi) [inline src/renderer/components/configure-recording-set-page/configure-recording-set.tsx:1]: Can you point out the changes in the content of this file?

File: src/renderer/components/configure-recording-set-page/index.tsx [hunk 3]
```diff
@@ -1,112 +1,44 @@
...
-  'デルタ式英語リストver5 (Delta English Ver. 5)',
-  'Z式CVVC-Normal (Z Chinese CVVC - Normal)',
-];
-
+import './index.scss';
+import { PreviewPage } from './preview-page';
+
+/*
+ Represents the page states controlled by configure-recording-set.
+ 'external' represents pages external from this system
```
Comment (iluminar-yi) [inline src/renderer/components/configure-recording-set-page/index.tsx:16]: What does "external from this system" mean?

File: src/renderer/components/configure-recording-set-page/index.tsx [hunk 4]
```diff
@@ -1,112 +1,44 @@
...
-  const [chosenKey = 'C', setChosenKey] = useLocalStorage<ScaleKey>(
-    getLSKey('ConfigureRecordingSetPage', 'chosenKey'),
-    'C',
-  );
-  const [chosenOctave = 3, setChosenOctave] = useLocalStorage<SupportedOctave>(
-    getLSKey('ConfigureRecordingSetPage', 'chosenOctave'),
-    3,
-  );
-  const [chosenName = '', setChosenName] = useLocalStorage(getLSKey('ConfigureRecordingSetPage', 'chosenName'), '');
+  const [recordingSetState, setRecordingSetState] = useState<RecordingPageState>('external');
```
Comment (iluminar-yi) [inline src/renderer/components/configure-recording-set-page/index.tsx]: Consider using `useLocalStorage` because that allows page state to be restored on macOS after the user clicks the "X"

File: src/renderer/components/configure-recording-set-page/index.tsx [hunk 5]
```diff
@@ -1,112 +1,44 @@
...
+  const [metadataStateIndex, setMetadataStateIndex] = useState<number>(-1);
+  const prevState = usePrevious(recordingSetState) ?? 'home';
+  const [transition, setTransition] = useState<boolean>(true);
+  const [filePathRec, setFilePathRec] = useState<string>('');
+  const [filePathOto, setFilePathOto] = useState<string>('');
+  const [filePathDvcfg, setFilePathDvcfg] = useState<string>('');
+
+  const [dropDownStates, setDropdownStateArray] = useState<MetadataState[]>([]);
+
+  const setDropdownState = () => {
```
Comment (iluminar-yi) [inline src/renderer/components/configure-recording-set-page/index.tsx]: `set` implies it is a setter. Maybe call it `goToNextDropdownState`?

File: src/renderer/components/configure-recording-set-page/index.tsx [hunk 6]
```diff
@@ -1,112 +1,44 @@
...
-  const [chosenOctave = 3, setChosenOctave] = useLocalStorage<SupportedOctave>(
-    getLSKey('ConfigureRecordingSetPage', 'chosenOctave'),
-    3,
-  );
-  const [chosenName = '', setChosenName] = useLocalStorage(getLSKey('ConfigureRecordingSetPage', 'chosenName'), '');
+  const [recordingSetState, setRecordingSetState] = useState<RecordingPageState>('external');
+  const [metadataStateIndex, setMetadataStateIndex] = useState<number>(-1);
+  const prevState = usePrevious(recordingSetState) ?? 'home';
+  const [transition, setTransition] = useState<boolean>(true);
+  const [filePathRec, setFilePathRec] = useState<string>('');
```
Comment (iluminar-yi) [inline src/renderer/components/configure-recording-set-page/index.tsx]: Let's add qualifying words in the front, i.e. `reclistFilePath` or `reclistPath`

File: src/renderer/components/configure-recording-set-page/index.tsx [hunk 7]
```diff
@@ -1,112 +1,44 @@
...
+  const [filePathOto, setFilePathOto] = useState<string>('');
```
Comment (iluminar-yi) [inline src/renderer/components/configure-recording-set-page/index.tsx]: `otoFilePath`

File: src/renderer/components/configure-recording-set-page/index.tsx [hunk 8]
```diff
@@ -1,112 +1,44 @@
...
+  const [filePathDvcfg, setFilePathDvcfg] = useState<string>('');
```
Comment (iluminar-yi) [inline src/renderer/components/configure-recording-set-page/index.tsx]: `dvcfgFilePath`

**Review by iluminar-yi — CHANGES_REQUESTED** [review]
Please update the PR title to match the format

File: src/renderer/components/configure-recording-set-page/index.scss [hunk 9]
```diff
@@ -0,0 +1,71 @@
+.slide-down-appear {
+  opacity: 0;
+  transform: translateY(-100%);
+}
+.slide-down-appear-active {
+  opacity: 1;
+  transform: translateY(0%);
+  transition: opacity 300ms, transform 300ms;
```
Comment (iluminar-yi) [inline src/renderer/components/configure-recording-set-page/index.scss:8]: Is duration still needed if it's also specified in the JS code (or vice versa)?

File: src/renderer/components/configure-recording-set-page/preview-page.tsx [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+        log.error(error);
+        throw error;
+      });
+    return () => {
+      isSubscribed = false;
+    };
+  });
+
+  return (
+    <div>
```
Comment (iluminar-yi) [inline src/renderer/components/configure-recording-set-page/preview-page.tsx]: Is `<div>` needed or is `<Fragment>` also fine?

File: src/renderer/components/configure-recording-set-page/preview-page.tsx [hunk 11]
```diff
@@ -0,0 +1,126 @@
...
+  };
+  useEffect(() => {
+    let isSubscribed = true;
+    if (!transition) {
+      setTransition(true);
+    }
+    (async (): Promise<'folder' | 'file' | false> => {
+      return checkFileExistence(fileName);
+    })()
+      .then((result) =>
```
Comment (iluminar-yi) [inline src/renderer/components/configure-recording-set-page/preview-page.tsx]: Wrap this in an async-await function to make the code more readable.

File: src/renderer/components/configure-recording-set-page/index.tsx [hunk 12]
```diff
@@ -115,154 +47,185 @@ const ConfigureRecordingSetPage: FC<{
...
-      await ensureFolderExists(recordingSetPath);
-      setRecordingSets([...recordingSets, newRecordingSet]);
-    }
-  };
 
-  const [selectedRecordingSetIndex = -1, setSelectedRecordingSetIndex] = useLocalStorage<number>(
-    getLSKey('ConfigureRecordingSetPage', 'selectedRecordingSetIndex'),
-    -1,
-  );
+  const updatePaths = (listName: string, isBuiltIn: boolean) => {
```
Comment (iluminar-yi) [inline src/renderer/components/configure-recording-set-page/index.tsx:64]: Can either `updatePaths` and `getFilePath` be extracted out of the component code?

File: src/renderer/components/configure-recording-set-page/index.tsx [hunk 13]
```diff
@@ -115,154 +47,185 @@ const ConfigureRecordingSetPage: FC<{
...
+    const getFilePath = (metaDataState: MetadataState, listName: string, isBuiltIn: boolean) => {
+      let filePath = '';
 
-  const removeRecordingSet = async (setToDelete: RecordingSet): Promise<void> => {
-    const willDelete = confirm(`Are you sure you want to delete ${setToDelete.name}?`);
-    if (willDelete) {
-      const newSets = recordingSets.filter((x) => x !== setToDelete);
-      setRecordingSets(newSets);
+      const stateToPathDeltaEnglish = {
```
Comment (iluminar-yi) [inline src/renderer/components/configure-recording-set-page/index.tsx]: These look like constants. Define them outside the component as:
```typescript
const DELTA_ENGLISH_STATE_FILEPATH_MAP = {
  'oto-ini': join('external', 'delta_eng_ver5', 'mkototemp.ini'),
  'list-preview': join('external', 'delta_eng_ver5', 'デルタ式engver5_reclist.txt'),
  dvcfg: join('external', 'To be added later'),
};
```

File: src/renderer/components/configure-recording-set-page/index.tsx [hunk 14]
```diff
@@ -115,154 +47,185 @@ const ConfigureRecordingSetPage: FC<{
...
+        return error;
+      }),
+      listPromise.catch((error) => {
+        return error;
+      }),
+      dvcfgPromise.catch((error) => {
+        return error;
+      }),
+    ])
+      .then((responses) => {
```
Comment (iluminar-yi) [inline src/renderer/components/configure-recording-set-page/index.tsx]: Use async-await

File: src/renderer/components/configure-recording-set-page/index.tsx [hunk 15]
```diff
@@ -115,154 +47,185 @@ const ConfigureRecordingSetPage: FC<{
...
+      return filePath;
+    };
+
+    let isSubscribed = true;
+
+    const otoFilePath = getFilePath('oto-ini', listName, isBuiltIn);
+    const listFilePath = getFilePath('list-preview', listName, isBuiltIn);
+    const dvcfgFilePath = getFilePath('dvcfg', listName, isBuiltIn);
+
+    const listPromise = checkFileExistence(listFilePath);
```
Comment (iluminar-yi) [inline src/renderer/components/configure-recording-set-page/index.tsx]: Still a lot of repetition here... Why not create an `async function checkMetadataFileExistence(metadataType: MetadataState, listName: string, isBuiltIn: boolean): Promise<boolean>`?

File: src/renderer/components/configure-recording-set-page/index.tsx [hunk 16]
```diff
@@ -115,154 +47,185 @@ const ConfigureRecordingSetPage: FC<{
...
-        recordingSets,
-      };
-      writeFile(
-        join(recordingProject.rootPath, PROJECT_CONFIG_FILENAME),
-        JSON.stringify(currentProjectState, null, 2),
-      ).catch(log.error);
+  switch (recordingSetState) {
+    case 'home':
+      return (
+        <ConfigureRecordingSet
```
Comment (iluminar-yi) [inline src/renderer/components/configure-recording-set-page/index.tsx]: This is the same as `case 'external':` it seems? You can collapse them. Or, if they are not the same, still consider collapsing them and make what's different clear.

File: src/renderer/components/configure-recording-set-page/index.tsx [hunk 17]
```diff
@@ -115,154 +47,185 @@ const ConfigureRecordingSetPage: FC<{
...
+          chosenBuiltInList={chosenBuiltInList}
+          rawSetChosenBuiltInList={rawSetChosenBuiltInList}
+          chosenCustomListPath={chosenCustomListPath}
+          rawSetChosenCustomListPath={rawSetChosenCustomListPath}
+          metaDataIndex={metadataStateIndex}
+          getFilePath={updatePaths}
+        />
+      );
+    case 'metadata':
+      if (dropDownStates[metadataStateIndex] == 'list-preview') {
```
Comment (iluminar-yi) [inline src/renderer/components/configure-recording-set-page/index.tsx]: Why not extract `dropDownStates[metadataStateIndex]` into a variable?

Comment (iluminar-yi) [reply src/renderer/components/configure-recording-set-page/index.tsx]: Also, this can be a switch statement too

File: src/renderer/components/configure-recording-set-page/preview-page.tsx [hunk 18]
```diff
@@ -0,0 +1,126 @@
...
+import { Consumer } from '../../types';
+import { checkFileExistence, readFile } from '../../utils';
+import ImageButton from '../image-button';
+
+import downButton from './down-button.svg';
+import switchPageButton from './switch-preview.svg';
+
+import { RecordingPageState } from '.';
+
+interface PreviewPageProps {
```
Comment (iluminar-yi) [inline src/renderer/components/configure-recording-set-page/preview-page.tsx:16]: Can you add documentations for the props?

File: src/renderer/components/configure-recording-set-page/preview-page.tsx [hunk 19]
```diff
@@ -0,0 +1,126 @@
...
+  const [className, setClassName] = useState<string>('slide-down');
+  const [pageText, setPageText] = useState<string>(pageName + ' not availible');
+
+  const transitionProps = {
+    in: transition,
+    appear: true,
+    timeout: 3000,
+    classNames: className,
+  };
+  useEffect(() => {
```
Comment (iluminar-yi) [inline src/renderer/components/configure-recording-set-page/preview-page.tsx]: Try to extract these `useEffect` out as custom hooks. By giving them a name, it can make the code more readable.

File: src/renderer/components/configure-recording-set-page/index.tsx [hunk 20]
```diff
@@ -1,112 +1,44 @@
...
-
-const BUILT_IN_RECORDING_LISTS: BuiltInRecordingList[] = [
-  'デルタ式英語リストver5 (Delta English Ver. 5)',
-  'Z式CVVC-Normal (Z Chinese CVVC - Normal)',
-];
-
+import './index.scss';
+import { PreviewPage } from './preview-page';
+
+/*
```
Comment (iluminar-yi) [inline src/renderer/components/configure-recording-set-page/index.tsx:14]: TS Docs have two `*` on the first line, like so:
```
/**
 * This is a TS Doc.
 */
```

Try typing just `/**` and press Enter and let VSCode generate the TS Doc block for you.

File: src/renderer/components/configure-recording-set-page/set-recording-list-configuration.tsx [hunk 21]
```diff
@@ -22,8 +23,18 @@ const SetRecordingListConfiguration: FC<SetRecordingListConfigurationProps> = ({
   setChosenBuiltInList,
   chosenCustomListPath,
   setChosenCustomListPath,
+  getFilePath,
 }) => {
   const { t } = useTranslation();
+  useEffect(() => {
```
Comment (iluminar-yi) [inline src/renderer/components/configure-recording-set-page/set-recording-list-configuration.tsx]: Extract into custom hook

File: src/renderer/components/configure-recording-set-page/set-recording-list-configuration.tsx [hunk 22]
```diff
@@ -32,7 +43,10 @@ const SetRecordingListConfiguration: FC<SetRecordingListConfigurationProps> = ({
             className={'w-100'}
             style={{ paddingRight: '0.75rem' }}
             value={chosenBuiltInList}
-            onChange={(e): void => setChosenBuiltInList(e.target.value)}>
+            testId={'test'}
```
Comment (iluminar-yi) [inline src/renderer/components/configure-recording-set-page/set-recording-list-configuration.tsx]: Is `testId` necessary? It appears to me that's the `escape hatch` provided by `react-test-library`. Can you do without it?

Comment (iluminar-yi) [reply src/renderer/components/configure-recording-set-page/set-recording-list-configuration.tsx]: Also, shouldn't the ID be more unique?

File: src/renderer/components/configure-recording-set-page/show-detail.svg [hunk 23]
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="170" height="66.258" viewBox="0 0 170 66.258"><defs><style>.a,.c{fill:none;stroke:#707070;}.a{font-size:30px;font-family:SourceHanSansCN-Regular, Source Han Sans CN;}.b{opacity:0.9;}</style></defs><g transform="translate(-910.5 -992.742)"><text class="a" transform="translate(995.5 1054)"><tspan x="-83.085" y="0">Show Detail</tspan></text><g class="b" transform="translate(1018 993.242) rotate(90)"><path class="c" d="M39.219,280.424,17.971,259.177a1.734,1.734,0,0,1,2.452-2.452l19.543,19.531,19.543-19.531a1.734,1.734,0,0,1,2.452,2.452L40.714,280.424a1.054,1.054,0,0,1-1.5,0Z" transform="translate(280.736 -17.466) rotate(90)"/></g></g></svg>
```
Comment (iluminar-yi) [inline src/renderer/components/configure-recording-set-page/show-detail.svg]: Can "Show Detail" be extracted out? I don't think it belongs in the icon

File: src/renderer/components/select/index.tsx [hunk 24]
```diff
@@ -4,6 +4,7 @@ import Form from 'react-bootstrap/Form';
 interface StrictSelectProps {
   value: string | number;
   onChange: React.ChangeEventHandler<HTMLSelectElement>;
+  testId?: string;
```
Comment (iluminar-yi) [inline src/renderer/components/select/index.tsx:10]: Can you document this field?

File: src/renderer/components/select/index.tsx [hunk 25]
```diff
@@ -13,13 +14,17 @@ interface SelectProps extends StrictSelectProps {
   style?: CSSProperties;
 }
 
-export const Select: FC<SelectProps> = ({ value, onChange, children, ...outerProps }) => {
+export const Select: FC<SelectProps> = ({ value, onChange, testId, children, ...outerProps }) => {
+  if (!testId) {
```
Comment (iluminar-yi) [inline src/renderer/components/select/index.tsx]: Use optional parameters, something like:
```
{ value, onChange, testId = '', children, ...outerProps }
```

File: src/renderer/types/global.d.ts [hunk 26]
```diff
@@ -1 +1,3 @@
 declare const __static: string;
+// Bug fix for ts error in react-use
```
Comment (iluminar-yi) [inline src/renderer/types/global.d.ts]: Can you be more specific, which hook caused this error? So that if that hook is ever deprecated, we know when to remove this.

File: src/renderer/index.scss [hunk 27]
```diff
@@ -36,3 +36,21 @@ td:empty::after,
 .place-held:empty::after {
   content: '\00a0';
 }
+// Global styles
```
Comment (iluminar-yi) [inline src/renderer/index.scss]: Why this change?

File: src/tests/ConfigureRecordingSetTests.test.tsx [hunk 28]
```diff
@@ -0,0 +1,176 @@
...
+
+import ConfigureRecordingSetPage from '../renderer/components/configure-recording-set-page/index';
+import { RecordingSet } from '../renderer/types';
+
+const flushPromises = () => new Promise(setImmediate);
+
+describe('configureRecordingSetPage', () => {
+  it('renders without showDetails button', async () => {
+    const originalError = console.error;
+    console.error = (...args: string[]) => {
```
Comment (iluminar-yi) [inline src/tests/ConfigureRecordingSetTests.test.tsx]: What is this for? And can this be extracted into a util just for tests?

File: src/tests/ConfigureRecordingSetTests.test.tsx [hunk 29]
```diff
@@ -0,0 +1,176 @@
...
+      if (/Warning.*not wrapped in act/.test(args[0])) {
+        return;
+      }
+      originalError.call(console, ...args);
+    };
+    i18next.use(initReactI18next);
```
Comment (iluminar-yi) [inline src/tests/ConfigureRecordingSetTests.test.tsx]: Is this working? I think I still see warnings

File: src/tests/ConfigureRecordingSetTests.test.tsx [hunk 30]
```diff
@@ -0,0 +1,176 @@
...
+
+import { cleanup, fireEvent, render, screen, waitFor } from '@testing-library/react';
+import i18next from 'i18next';
+import React from 'react';
+import { initReactI18next } from 'react-i18next/';
+
+import ConfigureRecordingSetPage from '../renderer/components/configure-recording-set-page/index';
+import { RecordingSet } from '../renderer/types';
+
+const flushPromises = () => new Promise(setImmediate);
```
Comment (iluminar-yi) [inline src/tests/ConfigureRecordingSetTests.test.tsx]: What is this for?

File: src/tests/ConfigureRecordingSetTests.test.tsx [hunk 31]
```diff
@@ -0,0 +1,176 @@
...
+        onBack={(_event: React.MouseEvent<HTMLElement, MouseEvent>) => {}}
+        onSetSelected={(_consumer: RecordingSet) => {}}
+      />,
+    );
+    let imageButtons = screen.getAllByRole('button');
+    expect(imageButtons).toHaveLength(4);
+    await flushPromises();
+    imageButtons = screen.getAllByRole('button');
+    expect(imageButtons).toHaveLength(4);
+    console.error = originalError;
```
Comment (iluminar-yi) [inline src/tests/ConfigureRecordingSetTests.test.tsx]: Shouldn't this be part of the `cleanup()`? Maybe create your own `cleanUp()` and wrap the library one inside of it?

File: src/tests/ConfigureRecordingSetTests.test.tsx [hunk 32]
```diff
@@ -0,0 +1,176 @@
+/**
```
Comment (iluminar-yi) [inline src/tests/ConfigureRecordingSetTests.test.tsx]: Move the file next to the source file, and name it just `<source-file-name>.test.tsx`

File: src/tests/ConfigureRecordingSetTests.test.tsx (hunk 31 above)
Comment (iluminar-yi) [reply src/tests/ConfigureRecordingSetTests.test.tsx]: Also, there should be `beforeEach` and `afterEach` (or the equivalent) provided by jest, just for this purpose

File: src/tests/ConfigureRecordingSetTests.test.tsx [hunk 33]
```diff
@@ -0,0 +1,176 @@
...
+  it('renders goes to oto-ini page after list preview page', async () => {
+    const originalError = console.error;
+    console.error = (...args: string[]) => {
+      if (/Warning.*not wrapped in act/.test(args[0])) {
+        return;
+      }
+      originalError.call(console, ...args);
+    };
+    expect(true).toBe(true);
+    // render(
```
Comment (iluminar-yi) [inline src/tests/ConfigureRecordingSetTests.test.tsx]: Remove commented out code

File: src/tests/ConfigureRecordingSetTests.test.tsx (hunk 32 above)
Comment (iluminar-yi) [reply src/tests/ConfigureRecordingSetTests.test.tsx]: But yes, thanks for adding new tests!

File: src/tests/ConfigureRecordingSetTests.test.tsx [hunk 34]
```diff
@@ -0,0 +1,176 @@
+/**
+ * @jest-environment jsdom
+ */
+/* eslint-disable @typescript-eslint/no-unused-vars */
```
Comment (iluminar-yi) [inline src/tests/ConfigureRecordingSetTests.test.tsx]: Are these linter suppression necessary?

**Review by iluminar-yi — CHANGES_REQUESTED** [review]

File: src/renderer/components/configure-recording-set-page/configure-recording-set.tsx (hunk 2 above)
Comment (joannalikespi) [reply src/renderer/components/configure-recording-set-page/configure-recording-set.tsx:1]: line 211 - addition of transition props for CSSTransitions
line 218 to end - addition of CSSTransition elements surrounding all HTML elements 
line 223 - call to set local state in onBack function
line 285 to 302 - changed show details button to be image button and added onClick functionality

File: src/renderer/components/configure-recording-set-page/index.tsx (hunk 3 above)
Comment (joannalikespi) [reply src/renderer/components/configure-recording-set-page/index.tsx:16]: The configure-recording-set page tracks a local state so it can determine which of it's subpages (the base configure-recording-set page or one of the preview metadata pages).  When the base configure-recording-set page loads it needs to know whether it's being loaded after one of the preview pages or from a page that isn't a local state of configure-recording-set (in order to determine whether to animate the transition in or not). The external state represents this second case

File: src/renderer/components/configure-recording-set-page/index.tsx (hunk 4 above)
Comment (joannalikespi) [reply src/renderer/components/configure-recording-set-page/index.tsx]: Should useLocalStorage always be used instead of useState? If not, when should it be used instead of useState?

File: src/renderer/components/configure-recording-set-page/show-detail.svg (hunk 23 above)
Comment (abbyabernathy) [reply src/renderer/components/configure-recording-set-page/show-detail.svg]: Zhou - I downloaded the 'Show Detail' svg directly off the mockup. So if you would like it to be different, can you tell me the best way to edit svgs or give me an svg that is what you are looking for?

File: src/tests/ConfigureRecordingSetTests.test.tsx (hunk 31 above)
Comment (joannalikespi) [reply src/tests/ConfigureRecordingSetTests.test.tsx]: The linter is currently set to not allow beforeEach or afterEach hooks - I can create my own before and after functions and call them before and after each test though

File: src/tests/ConfigureRecordingSetTests.test.tsx (hunk 30 above)
Comment (joannalikespi) [reply src/tests/ConfigureRecordingSetTests.test.tsx]: flushPromises forces the test to wait (with a timeout) to insure that all the promises that were created in a call return before the test moves on.  I use it though out the tests to make sure that everything actually finishes updating before i check for the changes

File: src/tests/ConfigureRecordingSetTests.test.tsx (hunk 34 above)
Comment (joannalikespi) [reply src/tests/ConfigureRecordingSetTests.test.tsx]: I think I can get rid of the first two, I do actually need the last two though - the empty function suppression is so i can pass empty functions into the component I'm testing (no need to mock them since they aren't being used), and the prefer-expect-assertions has to be disabled for react-testing-library to work with the async calls

File: src/renderer/components/configure-recording-set-page/index.tsx (hunk 12 above)
Comment (joannalikespi) [reply src/renderer/components/configure-recording-set-page/index.tsx:64]: I can extract getFilePath easily.
I originally kept updatePaths in index.tsx because it uses 5 different setState functions and I didn't want to have to pass those through three layers of components, but i could just create a single function that takes in values for and sets all the values and pass that single function through?

File: src/renderer/components/configure-recording-set-page/index.tsx (hunk 17 above)
Comment (joannalikespi) [reply src/renderer/components/configure-recording-set-page/index.tsx]: I'm using an array to make it easy to circle though the different local states - the preview page can just call the function that circularly increments the array instead of having to do more complex logic to determine what the next page is

File: src/renderer/components/configure-recording-set-page/configure-recording-set.tsx (hunk 2 above)
Comment (iluminar-yi) [reply src/renderer/components/configure-recording-set-page/configure-recording-set.tsx:1]: Thanks so much, next time feel free to just comment on those lines, should be easier for you

File: src/renderer/components/configure-recording-set-page/index.scss (hunk 9 above)
Comment (joannalikespi) [reply src/renderer/components/configure-recording-set-page/index.scss:8]: Yes, the transitions don't work if it's not specified in both locations

File: src/renderer/components/configure-recording-set-page/configure-recording-set.tsx [hunk 35]
```diff
@@ -0,0 +1,306 @@
...
+        recordingSets,
+      };
+      writeFile(
+        join(recordingProject.rootPath, PROJECT_CONFIG_FILENAME),
+        JSON.stringify(currentProjectState, null, 2),
+      ).catch(log.error);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [naiveSerialize(recordingProject), naiveSerialize(recordingSets)]);
+  const transitionProps = {
```
Comment (iluminar-yi) [inline src/renderer/components/configure-recording-set-page/configure-recording-set.tsx:211]: Empty line above

File: src/renderer/components/configure-recording-set-page/index.tsx (hunk 3 above)
Comment (iluminar-yi) [reply src/renderer/components/configure-recording-set-page/index.tsx:16]: Still a little confused. The second state doesn't refer to 'metadata'? Then what does 'metadata' represent?

File: src/renderer/components/configure-recording-set-page/index.tsx (hunk 4 above)
Comment (iluminar-yi) [reply src/renderer/components/configure-recording-set-page/index.tsx]: Almost always, as where the project stands right now. It's hard for you to test it because you don't develop on a mac, but it would be very obvious when it breaks. As a rule of thumb, any state variable that needs to be persisted to allow the component state to be restored when the page "reloads" (like clicking the refresh button on a normal webpage) needs to be persisted to the local storage.

That reminds me, you should be able to refresh the page using F5 key when the dev tool is open. Try refreshing the page and see if it breaks (or forgets everything you just did).

File: src/renderer/components/configure-recording-set-page/index.tsx (hunk 12 above)
Comment (iluminar-yi) [reply src/renderer/components/configure-recording-set-page/index.tsx:64]: Yeah that's an option, or wait until you do the other refactoring I mentioned below and there may be better ways to do it.

File: src/renderer/components/configure-recording-set-page/index.tsx (hunk 17 above)
Comment (iluminar-yi) [reply src/renderer/components/configure-recording-set-page/index.tsx]: I meant you could just do `const currentState = dropDownStates[metadataStateIndex]` above the statements.

File: src/renderer/components/configure-recording-set-page/show-detail.svg (hunk 23 above)
Comment (iluminar-yi) [reply src/renderer/components/configure-recording-set-page/show-detail.svg]: SVG is also XML-based, so it looks very similar to HTML if you format it. I removed `<text>` element and `.a` from the `<style>` element.
```
<svg xmlns="http://www.w3.org/2000/svg" width="170" height="66.258" viewBox="0 0 170 66.258">
    <defs>
        <style>.b{fill:none;stroke:#707070;}.a{opacity:0.9;}
        </style>
    </defs>
    <g transform="translate(-910.5 -992.742)">
        <g class="a" transform="translate(1018 993.242) rotate(90)">
            <path class="b"
                  d="M39.219,280.424,17.971,259.177a1.734,1.734,0,0,1,2.452-2.452l19.543,19.531,19.543-19.531a1.734,1.734,0,0,1,2.452,2.452L40.714,280.424a1.054,1.054,0,0,1-1.5,0Z"
                  transform="translate(280.736 -17.466) rotate(90)"/>
        </g>
    </g>
</svg>
```

File: src/tests/ConfigureRecordingSetTests.test.tsx (hunk 34 above)
Comment (iluminar-yi) [reply src/tests/ConfigureRecordingSetTests.test.tsx]: Empty function -> There is a `noOp()` function I defined in one of the utils.

Other things sound good.

File: src/tests/ConfigureRecordingSetTests.test.tsx (hunk 31 above)
Comment (iluminar-yi) [reply src/tests/ConfigureRecordingSetTests.test.tsx]: Interesting. What are the reasons it gives?

File: src/tests/ConfigureRecordingSetTests.test.tsx (hunk 30 above)
Comment (iluminar-yi) [reply src/tests/ConfigureRecordingSetTests.test.tsx]: OK, in that case, maybe put this in its own `test-utils.ts`

**Review by iluminar-yi — CHANGES_REQUESTED** [review]

File: src/renderer/components/configure-recording-set-page/index.tsx (hunk 3 above)
Comment (joannalikespi) [reply src/renderer/components/configure-recording-set-page/index.tsx:16]: apologies, I forgot that I split off the states into two separate types of state.  
There's three 'primary' sub-states: home, external, and metadata.  Home refers to the configure-recording-set base page, external refers to a page outside of the configure-recording-set subsystem, and metadata, which refers to the metadata  preview pages (oto/reclist/dvcfg).  
Since it is determined at runtime which preview pages are able to be accessed (since they do not show up if the file does not exist in the reclist's folder), i needed to be able to iterate though whichever pages existed when the side arrow button was clicked.  In order to do this, i separated the states that could potentially get iterated through into their own type of state with a different type of setter.  So basically, as soon as the "show details" button is clicked, the RecordingPageState is set to metadata, and the MetadataState is what's used to determine what is rendered until the state is set back to home;
When rendering the configure-recording-set base page, it either needs to be told to use an animated transition (if it's coming back from one of the metadata pages) or to appear normally (if it's coming from any other page) When either the "next" or "back" button gets clicked, it sets the state to external so that when we go back to the configure-recording-set base page, it knows not to use the animated transition

File: src/renderer/components/configure-recording-set-page/set-recording-list-configuration.tsx (hunk 22 above)
Comment (joannalikespi) [reply src/renderer/components/configure-recording-set-page/set-recording-list-configuration.tsx]: I changed the ID to make it more unique, but unfortunately it is necessary to make the dropdown selectable in the tests I'm running.  I need to select it/feed input to it in order to run the tests, and because it doesn't have default text (and there's another selector of the same type on the screen) i need the ID to make it identifyable

File: src/renderer/components/configure-recording-set-page/configure-recording-set.tsx (hunk 35 above)
Comment (iluminar-yi) [reply src/renderer/components/configure-recording-set-page/configure-recording-set.tsx:211]: Bump

File: src/renderer/components/configure-recording-set-page/index.tsx (hunk 20 above)
Comment (iluminar-yi) [reply src/renderer/components/configure-recording-set-page/index.tsx:14]: Bump

File: src/renderer/components/configure-recording-set-page/index.tsx (hunk 3 above)
Comment (iluminar-yi) [reply src/renderer/components/configure-recording-set-page/index.tsx:16]: I see. Thank you!

File: src/renderer/components/configure-recording-set-page/index.tsx [hunk 36]
```diff
@@ -1,112 +1,57 @@
...
-    if (projectFile.recordingSets.length) {
-      // FIXME Need real loading
-      setRecordingSets(projectFile.recordingSets);
-    }
-  }, [projectFile, setRecordingSets]);
 
-  const [chosenKey = 'C', setChosenKey] = useLocalStorage<ScaleKey>(
-    getLSKey('ConfigureRecordingSetPage', 'chosenKey'),
-    'C',
+  const [dropDownStates = new Array<MetadataState>(), setDropdownStateArray] = useLocalStorage(
```
Comment (iluminar-yi) [inline src/renderer/components/configure-recording-set-page/index.tsx]: I think it's more concise to say
```typescript
const [dropDownStates = [], setDropdownStateArray] = useLocalStorage<Array<MetadataState>>(getLSKey('ConfigureRecordingSetPage', 'dropDownStates'), []);
```

File: src/renderer/components/configure-recording-set-page/preview-page.tsx (hunk 18 above)
Comment (iluminar-yi) [reply src/renderer/components/configure-recording-set-page/preview-page.tsx:16]: Actually, field TS Doc should go on the fields themselves:
```
interface PreviewPageProps {
  /**
   * Sets the local pagestate (controls which subpage is displayed)
   */
  setRecordingSetState: Consumer<RecordingPageState>;
  /**
   * boolean to determin whether the animated page in transition has occured
   */
  pageName: string;
  // ...
}

```

File: src/renderer/components/configure-recording-set-page/preview-page.tsx (hunk 19 above)
Comment (iluminar-yi) [reply src/renderer/components/configure-recording-set-page/preview-page.tsx]: Bump

**Review by iluminar-yi — CHANGES_REQUESTED** [review]

**Comment by iluminar-yi** [comment]
I also updated the title again:
`feat:` is a keyword that must not be capitalized.
Also, use present tense for commit messages so I changed `Animated` to `animate`

**Review by iluminar-yi — APPROVED** [review]
